hls4ml/optimization/keras/reduction.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+from tensorflow.keras.layers import Conv2D, Dense
+from tensorflow.keras.models import Sequential
+
+from hls4ml.optimization.keras.utils import get_last_layer_with_weights
+
+'''
+Function for removing zero neurons & filters from a model and rewiring the model graph
+This function is built on top of Keras Surgeon available at: https://github.com/BenWhetton/keras-surgeon
+Keras Surgeon is no longer under active development and does not work for TensorFlow 2.3+ and QKeras
+The baseline version was forked and updated, available at: https://github.com/bo3z/keras-surgeon
+
+Args:
+    - model (keras.model): Input model
+
+Return:
+    - reduced (keras.model): Modified model, with redundant structures removed
+
+    '''
+
+
+def reduce_model(model):
+    # TODO - Should we make Keras Surgeon a hard requirement in setup.cfg? If so, needs to be installed from git, @bo3z fork
```

Comment:
Yes, and I would transfer it to `fastmachinelearning/keras-surgeon` if you agree @bo3z (you can still be maintainer of course).

Reply:
Done.